### PR TITLE
fix SMS provider huawei

### PR DESCRIPTION
**Pull request by zhu733756**
Signed-off-by: zhu733756 <talonzhu@yunify.com>
- remove the field TemplateParas, the value of this field should be generated by the notification messages.
- fix a bug in the generation process of the request header.
- fix templateParas: add regex rules to extract the return values of alertname,alerttype,severity,message,summary

### CODE REVIEW

File: pkg/notify/notifier/sms/huawei.go [hunk 1]
```diff
@@ -113,9 +114,38 @@ func (h *HuaweiNotifier) MakeRequest(ctx context.Context, messages string) error
...
+	sha256 := getSha256Code(nonce + now + strings.ReplaceAll(appSecret, "\n", ""))
 	passwordDigest := base64.StdEncoding.EncodeToString([]byte(sha256))
-	return fmt.Sprintf(`"UsernameToken Username="%s",PasswordDigest="%s",Nonce="%s",Created="%s"`, appKey, passwordDigest, nonce, now)
+	return fmt.Sprintf(`UsernameToken Username="%s",PasswordDigest="%s",Nonce="%s",Created="%s"`, strings.ReplaceAll(appKey, "\n", ""), passwordDigest, nonce, now)
+}
+
+// Note here, the keys corresponding to the return values are alertname,alerttype,severity,message,summary.
+// Pls make your custem SMS template containing five placeholders, you can set the template on the console of SMS huawei cloud provider.
+// i.e: Reviced notifications: alertname:${TEXT}, alerttype:${TEXT}, serverity:${TEXT}, message:${message}, summary:${summary}
+func generateParas(messages string) []string {
```
Comment (benjaminhuo) [inline pkg/notify/notifier/sms/huawei.go:125]: generateParas =>  generateParams

File: pkg/notify/notifier/sms/huawei.go [hunk 2]
```diff
@@ -113,9 +114,38 @@ func (h *HuaweiNotifier) MakeRequest(ctx context.Context, messages string) error
...
-	sha256 := getSha256Code(nonce + now + appSecret)
+	sha256 := getSha256Code(nonce + now + strings.ReplaceAll(appSecret, "\n", ""))
 	passwordDigest := base64.StdEncoding.EncodeToString([]byte(sha256))
-	return fmt.Sprintf(`"UsernameToken Username="%s",PasswordDigest="%s",Nonce="%s",Created="%s"`, appKey, passwordDigest, nonce, now)
+	return fmt.Sprintf(`UsernameToken Username="%s",PasswordDigest="%s",Nonce="%s",Created="%s"`, strings.ReplaceAll(appKey, "\n", ""), passwordDigest, nonce, now)
+}
+
+// Note here, the keys corresponding to the return values are alertname,alerttype,severity,message,summary.
+// Pls make your custem SMS template containing five placeholders, you can set the template on the console of SMS huawei cloud provider.
+// i.e: Reviced notifications: alertname:${TEXT}, alerttype:${TEXT}, serverity:${TEXT}, message:${message}, summary:${summary}
```
Comment (benjaminhuo) [inline pkg/notify/notifier/sms/huawei.go:124]: Reviced => Received

File: pkg/notify/notifier/sms/huawei.go [hunk 3]
```diff
@@ -113,9 +114,38 @@ func (h *HuaweiNotifier) MakeRequest(ctx context.Context, messages string) error
...
 	nonce := strings.ReplaceAll(string(uuid.NewUUID()), "-", "")
-	sha256 := getSha256Code(nonce + now + appSecret)
+	sha256 := getSha256Code(nonce + now + strings.ReplaceAll(appSecret, "\n", ""))
 	passwordDigest := base64.StdEncoding.EncodeToString([]byte(sha256))
-	return fmt.Sprintf(`"UsernameToken Username="%s",PasswordDigest="%s",Nonce="%s",Created="%s"`, appKey, passwordDigest, nonce, now)
+	return fmt.Sprintf(`UsernameToken Username="%s",PasswordDigest="%s",Nonce="%s",Created="%s"`, strings.ReplaceAll(appKey, "\n", ""), passwordDigest, nonce, now)
+}
+
+// Note here, the keys corresponding to the return values are alertname,alerttype,severity,message,summary.
+// Pls make your custem SMS template containing five placeholders, you can set the template on the console of SMS huawei cloud provider.
```
Comment (benjaminhuo) [inline pkg/notify/notifier/sms/huawei.go:123]: Pls make your custem SMS template containing five placeholders =>
Pls make your custom SMS template containing five placeholders

Comment (benjaminhuo) [inline pkg/notify/notifier/sms/huawei.go:123]: you can set the template on the console of SMS huawei cloud provider. =>
you can create an SMS template in Huawei Cloud's SMS console.

**Review by benjaminhuo — CHANGES_REQUESTED** [review]

**Review by wanjunlei — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -464,6 +464,107 @@ EOF
...
+  labels:
+    app: notification-manager
+  name: default-sms-secret
+type: Opaque
+```
+For SMS templates,  you can create them in your SMS provider's SMS console.
+For Huawei Cloud SMS provider, pls make your custom SMS template containing five placeholders. 
+For a detailed description, if you have a template like this: 
+```
+[QingCloud Container platform alarm] alertname = ${TEXT}, alerttype = ${TEXT}, severity = ${TEXT}, message = ${TEXT}, summary = ${TEXT}
```
Comment (benjaminhuo) [inline README.md:561]: QingCloud Container platform alarm =>
KubeSphere alerts

Comment (zhu733756) [reply README.md]: fixed

**Review by benjaminhuo — APPROVED** [review]

**Pull request merged**